### PR TITLE
podvm: contribute podvm-generic-ubuntu oci images

### PR DESCRIPTION
**Pull request by liudalibj**
- download the built podvm-generic-ubuntu oci images from cos
- push them to quay.io with caa_commitid as tag

fixes: #732
for https://github.com/confidential-containers/cloud-api-adaptor/issues/732#issuecomment-1560646723

### CODE REVIEW

**Comment by liudalibj** [comment]
I tested the approach at https://github.com/liudalibj/cloud-api-adaptor/actions/runs/5077214913
the debug codes push the built `generic-ubuntu` oci images to my private repos. A step `Download and Push podvm ubutnu oci image` is added, the core logs:
```
...
Successfully downloaded 'podvm-generic-ubuntu-s390x-2536ae8ef62e5aed5cda224a428d389e52b68768.tar' from bucket 'daily-e2e-test-bucket'
617.69 MiB downloaded.
Loaded image: quay.io/repository/confidential-containers/podvm-generic-ubuntu-s390x:2536ae8ef62e5aed5cda224a428d389e52b68768
...
The push refers to repository [quay.io/***/podvm-generic-ubuntu-s390x]
a3349ed54318: Preparing
a3349ed54318: Pushed
2536ae8ef62e5aed5cda224a428d389e52b68768: digest: sha256:6f38b144a5209e94ef6cd6498fdd55bebbb01386ceba84dfdb870e5db93245c0 size: 530
```

**Comment by stevenhorsman** [comment]
> I tested the approach at https://github.com/liudalibj/cloud-api-adaptor/actions/runs/5077214913 the debug codes push the built `generic-ubuntu` oci images to my private repos. A step `Download and Push podvm ubutnu oci image` is added, the core logs:
> 
> ```
> ...
> Successfully downloaded 'podvm-generic-ubuntu-s390x-2536ae8ef62e5aed5cda224a428d389e52b68768.tar' from bucket 'daily-e2e-test-bucket'
> 617.69 MiB downloaded.
> Loaded image: quay.io/repository/confidential-containers/podvm-generic-ubuntu-s390x:2536ae8ef62e5aed5cda224a428d389e52b68768
> ...
> The push refers to repository [quay.io/***/podvm-generic-ubuntu-s390x]
> a3349ed54318: Preparing
> a3349ed54318: Pushed
> 2536ae8ef62e5aed5cda224a428d389e52b68768: digest: sha256:6f38b144a5209e94ef6cd6498fdd55bebbb01386ceba84dfdb870e5db93245c0 size: 530
> ```
> 
> I didn't update the `latest` tag. Only use `caa_commit_id` as the tag.

Quick question before I start the review. I can't see the test image in https://quay.io/repository/confidential-containers/podvm-generic-ubuntu-s390x?tab=tags - did you push it to your own account for testing, or am I looking in the wrong place?

**Comment by liudalibj** [comment]
> > I tested the approach at https://github.com/liudalibj/cloud-api-adaptor/actions/runs/5077214913 the debug codes push the built `generic-ubuntu` oci images to my private repos. A step `Download and Push podvm ubutnu oci image` is added, the core logs:
> > ```
> > ...
> > Successfully downloaded 'podvm-generic-ubuntu-s390x-2536ae8ef62e5aed5cda224a428d389e52b68768.tar' from bucket 'daily-e2e-test-bucket'
> > 617.69 MiB downloaded.
> > Loaded image: quay.io/repository/confidential-containers/podvm-generic-ubuntu-s390x:2536ae8ef62e5aed5cda224a428d389e52b68768
> > ...
> > The push refers to repository [quay.io/***/podvm-generic-ubuntu-s390x]
> > a3349ed54318: Preparing
> > a3349ed54318: Pushed
> > 2536ae8ef62e5aed5cda224a428d389e52b68768: digest: sha256:6f38b144a5209e94ef6cd6498fdd55bebbb01386ceba84dfdb870e5db93245c0 size: 530
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I didn't update the `latest` tag. Only use `caa_commit_id` as the tag.
> 
> Quick question before I start the review. I can't see the test image in https://quay.io/repository/confidential-containers/podvm-generic-ubuntu-s390x?tab=tags - did you push it to your own account for testing, or am I looking in the wrong place?

As I use a fork but not a branch under 'https://github.com/confidential-containers/cloud-api-adaptor', so I pushed to my own account for testing.
image
- quay.io/liudalibj/podvm-generic-ubuntu-s390x-se
- quay.io/liudalibj/podvm-generic-ubuntu-s390x
- quay.io/liudalibj/podvm-generic-ubuntu-amd64

I removed the test codes, so in current pr codes, it will push to (not happen yet)
- quay.io/confidential-containers/podvm-generic-ubuntu-s390x-se
- quay.io/confidential-containers/podvm-generic-ubuntu-s390x
- quay.io/confidential-containers/podvm-generic-ubuntu-amd64

**Comment by liudalibj** [comment]
I make the test repos public:
- https://quay.io/repository/liudalibj/podvm-generic-ubuntu-amd64?tab=history
- https://quay.io/repository/liudalibj/podvm-generic-ubuntu-s390x?tab=history
- https://quay.io/repository/liudalibj/podvm-generic-ubuntu-s390x-se?tab=history
@stevenhorsman you can check again.

**Review by stevenhorsman — COMMENTED** [review]
Minor typos

File: .github/workflows/daily-e2e-tests-ibmcloud.yaml [hunk 1]
```diff
@@ -64,5 +64,38 @@ jobs:
...
+          echo ${caa_commitid} > caa_commitid
+        env:
+          bucket_name : "daily-e2e-test-bucket"
+      - name: Login to Quay container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+      - name: Download and Push podvm ubutnu oci image
```
[the remark this answers is not in the document]
Comment (liudalibj) [reply .github/workflows/daily-e2e-tests-ibmcloud.yaml:77]: updated

File: .github/workflows/daily-e2e-tests-ibmcloud.yaml [hunk 2]
```diff
@@ -64,5 +64,38 @@ jobs:
...
+      - name: Login to Quay container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+      - name: Download and Push podvm ubuntu oci image
+        run: |
+          caa_commit_id=$(cat caa_commitid)
+          echo "The build podvm image is base on CAA commit_id: ${caa_commit_id}"
```
Comment (bpradipt) [inline .github/workflows/daily-e2e-tests-ibmcloud.yaml:80]:
```suggestion
echo "The built podvm image is based on CAA commit_id: ${caa_commit_id}"
```

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @liudalibj. 
I added a minor suggestion

File: .github/workflows/daily-e2e-tests-ibmcloud.yaml (hunk 2 above)
Comment (liudalibj) [reply .github/workflows/daily-e2e-tests-ibmcloud.yaml]: updated, thanks @bpradipt !

**Comment by huoqifeng** [comment]
> LGTM - I just need to remember to start using them rather than building my own podvm images every time! Thanks @liudalibj

Was the oci image's usage documented somewhere?

**Comment by liudalibj** [comment]
> > LGTM - I just need to remember to start using them rather than building my own podvm images every time! Thanks @liudalibj
> 
> Was the oci image's usage documented somewhere?

Not yet, we should update related documents in future pr, after monitor the workflow for some days and it works as expected.

**Pull request merged**